src/Core/Managed/Shared/DataContracts/RequestTelemetry.cs

```diff
@@ -223,19 +223,31 @@ void ITelemetry.Sanitize()
             // Required field
             if (!this.Success.HasValue)
             {
-                int responseCode;
-
-                if (int.TryParse(this.ResponseCode, NumberStyles.Any, CultureInfo.InvariantCulture, out responseCode))
-                {
-                    this.Success = (responseCode < 400) || (responseCode == 401);
-                }
-                else
-                {
-                    this.Success = true;
-                }
+                this.Success = this.GetSuccessFromResponseCode();
             }
 
             this.Context.SanitizeTelemetryContext();
         }
+
+        internal bool GetSuccessFromResponseCode()
```

Comment:
@SergeyKanzhelev , refactoring this is not in scope for this particular change. We can do this separately, but right now I need to do a minimal, non-impacting change to the SDK that will unblock my progress. This PR is exactly that. All I did was to move a few lines into a separate method so that I can reuse them.